zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -1599,8 +1599,11 @@ public void processPacket(ServerCnxn cnxn, ByteBuffer incomingBuffer) throws IOE
                 }
             }
             if (authReturn == KeeperException.Code.OK) {
+                ZooKeeperSaslServer saslServer = cnxn.zooKeeperSaslServer;
                 LOG.debug("Authentication succeeded for scheme: {}", scheme);
-                LOG.info("auth success {}", cnxn.getRemoteSocketAddress());
+                LOG.info("Session 0x{}: auth success for authorizationID={}/{}",
+                        Long.toHexString(cnxn.getSessionId()), saslServer.getAuthorizationID(),
+                        cnxn.getRemoteSocketAddress());
```

Comment:
That's right @ztzg.
I was just helping clear out the backlog when I took this up but I think I'll need to add in more code so took a breather on this.
I would still like to give it a shot till the weekend if you don't mind. Will tag you for updated codes' review :)

Reply:
Sure; no hurry/pressure.  Was just checking as I was working in the same area…

Reply:
@ztzg I added some code to protect the use of sasl in the log statement only if its not null.
Not sure if that is sufficient / appropriate.. please advise!

Reply:
Hi @Ghatage,

No, I don't think that's what we want—sorry for not being clearer in my previous comment.

This is the code path where "traditional" auth packets are being handled. While it is technically possible to send a `sasl` auth packet to a server, the [`handleAuthentication` method](https://github.com/apache/zookeeper/blob/fe940cdd8fb23ba09684cefb73233d570f4a20fa/zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/SASLAuthenticationProvider.java#L34) of `SASLAuthenticationProvider` always returns `AUTHFAILED`.  So you'll *never* enter that block in response to SASL auth.

Now, while I think it would be good to add at least the `sessionId` and `scheme` to that message, adding the "user name" corresponding to a traditional `AuthenticationProvider` is quite a bit more complicated—and doing it "very cleanly" would arguably require incompatible infrastructure changes.

(For example, `DigestAuthenticationProvider` can add [up to two `Id`s](https://github.com/apache/zookeeper/blob/fe940cdd8fb23ba09684cefb73233d570f4a20fa/zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/DigestAuthenticationProvider.java#L99-L102) to a connection, and the existing interface does not allow it to return them; you'd have to "diff" the before and after `Id` sets to know what has changed.)

So I'm not sure it's worth the effort, given that everybody is moving towards SASL and/or X509 anyway.  And even if one wants to keep using a traditional provider, it is usually more useful to let it do its own (domain-specific) logging.

In conclusion: I'd just add the `sessionId` and `scheme` there, and ignore the actual `Id`(s).

Hope this helps, -D

Reply:
> In conclusion: I'd just add the sessionId and scheme there, and ignore the actual Id(s).

make sense to me.
(for SASL I think we already have log outputs about the authenticated principal names)